File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
       vector<int> dp(nums.size(),-1);

        //base case
        dp[nums.size()-1] = 1;

       for(int idx=nums.size()-2; idx>=0; idx--){
            bool jump=false;
            for(int i=1; i<=nums[idx]; i++){
            jump = jump || dp[idx+i];
            }
            dp[idx] = jump;
       }
       return dp[0];
    }
};

File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        goal = len(nums) - 1 
        for currpos in range(len(nums)-2,-1,-1):
            if(currpos + nums[currpos] >= goal):
                goal = currpos

        return True if goal == 0 else False